executor/adapter.go

```diff
@@ -243,7 +243,7 @@ func (a *ExecStmt) Exec(ctx context.Context) (sqlexec.RecordSet, error) {
 
 	var txnStartTS uint64
 	if sctx.Txn(false).Valid() {
-		txnStartTS = sctx.Txn().StartTS()
+		txnStartTS = sctx.Txn(true).StartTS()
```

Comment:
Can you add some comments here, to explain why we need to pass a param to Txn()?